lib/active_record/connection_adapters/oracle_enhanced/schema_statements.rb

```diff
@@ -482,19 +482,23 @@ def new_column_from_field(table_name, field)
               field["data_default"] = false if (field["data_default"] == "N" && OracleEnhancedAdapter.emulate_booleans_from_strings)
             end
 
-            type_metadata = fetch_type_metadata(field["sql_type"])
+            type_metadata = fetch_type_metadata(field["sql_type"], is_virtual)
+            # Somehow, `type_metadata` does not have `virtual` and its class is `ActiveRecord::ConnectionAdapters::SqlTypeMetadata`
+            # Not ActiveRecord::ConnectionAdapters::OracleEnhanced::TypeMetadata`
+            require "pp"
+            pp type_metadata if is_virtual
```

Comment:
Can you confirm `p type_metadata.class`?
`pp` calls `object.pretty_print`, so `DelegateClass` will delegate it to the instance of `SqlTypeMetadata` in `OracleEnhanced::TypeMetadata`.

Reply:
Thanks for the answer.

`p type_metadata.class if is_virtual` shows
`ActiveRecord::ConnectionAdapters::OracleEnhanced::TypeMetadata` as expected, `p type_metadata.inspec` shows `"#<ActiveRecord::ConnectionAdapters::SqlTypeMetadata:0x0000000003a1f8f8 @sql_type=\"VARCHAR2(512)\", @type=:string, @limit=512, @precision=nil, @scale=nil>"`, I do not know if `object#inspect` is affected by delegate or not but I wanted it shows `virtual`
Maybe something wrong with `fetch_type_metadata` method itself.  Will investigate.

Reply:
I think `fetch_type_metadata` is working fine now. All specs are green.